pkgs/http/lib/src/base_request.dart

```diff
@@ -89,14 +89,33 @@ abstract class BaseRequest {
   bool _finalized = false;
 
   static final _tokenRE = RegExp(r"^[\w!#%&'*+\-.^`|~]+$");
+
   static String _validateMethod(String method) {
     if (!_tokenRE.hasMatch(method)) {
       throw ArgumentError.value(method, 'method', 'Not a valid method');
     }
     return method;
   }
 
-  BaseRequest(String method, this.url)
+  /// On upload progress callback.
+  ///
+  /// If defined, this callback will be called when the upload progress changes.
+  ///
+  /// In browser, uses XMLHttpRequest's "xhr.upload.onLoad" event.
```

Comment:
Should we send more than a single event using `progress`?

Reply:
I didn't understand exactly what you mean.  I am sorry.

Reply:
The `xhr.upload.onLoad` event will fire exactly once. This will not show incremental progress.

The `xhr.upload.progress` event may fire multiple times.

Reply:
Done